test/integration/client/statement_timeout-tests.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+var helper = require('./test-helper')
+var Client = helper.Client
+
+var suite = new helper.Suite()
+
+var conInfo = helper.config
+
+function getConInfo (override) {
+  return Object.assign({}, conInfo, override )
+}
+
+function getStatementTimeout (conf, cb) {
+  var client = new Client(conf)
+  client.connect(assert.success(function () {
+    client.query('SHOW statement_timeout', assert.success(function (res) {
+      var statementTimeout = res.rows[0].statement_timeout
+      cb(statementTimeout)
+      client.end()
+    }))
+  }))
+}
+
+if (!helper.args.native) { // statement_timeout is not supported with the native client
+  suite.test('No default statement_timeout ', function (done) {
+    getConInfo()
+    getStatementTimeout({}, function (res) {
+      assert.strictEqual(res, '0') // 0 = no timeout
+      done()
+    })
+  })
+
+  suite.test('statement_timeout integer is used', function (done) {
+    var conf = getConInfo({
+      'statement_timeout': 3000
+    })
+    getStatementTimeout(conf, function (res) {
+      assert.strictEqual(res, '3s')
+      done()
+    })
+  })
+
+  suite.test('statement_timeout float is used', function (done) {
+    var conf = getConInfo({
+      'statement_timeout': 3000.7
+    })
+    getStatementTimeout(conf, function (res) {
+      assert.strictEqual(res, '3s')
+      done()
+    })
+  })
+
+  suite.test('statement_timeout string is used', function (done) {
+    var conf = getConInfo({
+      'statement_timeout': '3000'
+    })
+    getStatementTimeout(conf, function (res) {
+      assert.strictEqual(res, '3s')
+      done()
+    })
+  })
+}
```

Comment:
It looks like the tests only check for the correct configuration of statement_timeout.
I think we miss a test for checking if the timeout is actually used, by using a query with `pg_sleep(whicheverValueAboveTimeout)`

Reply:
Sure. But isn't that testing Postgres itself? The tests already confirmed that the setting is set on the connection.

Reply:
> Sure. But isn't that testing Postgres itself? The tests already confirmed that the setting is set on the connection.

You may have a point here, but the test seems pretty simple to write, so IMHO it's better to add it.
Whatsoever @brianc has the final word.

Reply:
I've written it. Travis is running so I'm hoping the error response from postgres is the same from all the versions you guys support. I asserted the error code only with the hope that is will be less fragile than expecting a particular error message.